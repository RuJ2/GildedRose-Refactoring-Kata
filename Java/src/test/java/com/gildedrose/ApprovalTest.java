/**
 * @(#)ApprovalTest.java, 8æœˆ 09, 2023.
 * <p>
 * Copyright 2023 fenbi.com. All rights reserved.
 * FENBI.COM PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */
package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

/**
 * @author rujiapei
 */
public class ApprovalTest {

    @Test
    public void updateQuality(){
        CombinationApprovals.verifyAllCombinations(
            this::callUpdateQuality,
            new String[]{"a common item", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros", "Conjured Mana Cake"},
            new Integer[]{0, 11, -1},
            new Integer[]{0, 1, -1, 49, 50, 55}
        );
    }

    @Test
    public String callUpdateQuality(String name, int sellIn, int quality){
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }
}
