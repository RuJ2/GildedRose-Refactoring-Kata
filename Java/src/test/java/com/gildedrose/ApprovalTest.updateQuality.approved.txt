[a common item, 0, 0] => a common item, -1, 0 
[a common item, 0, 1] => a common item, -1, 0 
[a common item, 0, -1] => a common item, -1, 0 
[a common item, 0, 49] => a common item, -1, 47 
[a common item, 0, 50] => a common item, -1, 48 
[a common item, 0, 55] => a common item, -1, 53 
[a common item, 11, 0] => a common item, 10, 0 
[a common item, 11, 1] => a common item, 10, 0 
[a common item, 11, -1] => a common item, 10, 0 
[a common item, 11, 49] => a common item, 10, 48 
[a common item, 11, 50] => a common item, 10, 49 
[a common item, 11, 55] => a common item, 10, 54 
[a common item, -1, 0] => a common item, -2, 0 
[a common item, -1, 1] => a common item, -2, 0 
[a common item, -1, -1] => a common item, -2, 0 
[a common item, -1, 49] => a common item, -2, 47 
[a common item, -1, 50] => a common item, -2, 48 
[a common item, -1, 55] => a common item, -2, 53 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, -1] => Aged Brie, -1, 1 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 0, 55] => Aged Brie, -1, 55 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, -1] => Aged Brie, 10, 0 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Aged Brie, 11, 55] => Aged Brie, 10, 55 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, -1] => Aged Brie, -2, 1 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, -1, 55] => Aged Brie, -2, 55 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, -1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 55] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, -1] => Backstage passes to a TAFKAL80ETC concert, 10, 0 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 55] => Backstage passes to a TAFKAL80ETC concert, 10, 55 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, -1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 55] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros, 0, 1 
[Sulfuras, Hand of Ragnaros, 0, -1] => Sulfuras, Hand of Ragnaros, 0, -1 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 55] => Sulfuras, Hand of Ragnaros, 0, 55 
[Sulfuras, Hand of Ragnaros, 11, 0] => Sulfuras, Hand of Ragnaros, 11, 0 
[Sulfuras, Hand of Ragnaros, 11, 1] => Sulfuras, Hand of Ragnaros, 11, 1 
[Sulfuras, Hand of Ragnaros, 11, -1] => Sulfuras, Hand of Ragnaros, 11, -1 
[Sulfuras, Hand of Ragnaros, 11, 49] => Sulfuras, Hand of Ragnaros, 11, 49 
[Sulfuras, Hand of Ragnaros, 11, 50] => Sulfuras, Hand of Ragnaros, 11, 50 
[Sulfuras, Hand of Ragnaros, 11, 55] => Sulfuras, Hand of Ragnaros, 11, 55 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 1] => Sulfuras, Hand of Ragnaros, -1, 1 
[Sulfuras, Hand of Ragnaros, -1, -1] => Sulfuras, Hand of Ragnaros, -1, -1 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, -1, 55] => Sulfuras, Hand of Ragnaros, -1, 55 
[Conjured Mana Cake, 0, 0] => Conjured Mana Cake, -1, 0 
[Conjured Mana Cake, 0, 1] => Conjured Mana Cake, -1, 0 
[Conjured Mana Cake, 0, -1] => Conjured Mana Cake, -1, 0 
[Conjured Mana Cake, 0, 49] => Conjured Mana Cake, -1, 45 
[Conjured Mana Cake, 0, 50] => Conjured Mana Cake, -1, 46 
[Conjured Mana Cake, 0, 55] => Conjured Mana Cake, -1, 51 
[Conjured Mana Cake, 11, 0] => Conjured Mana Cake, 10, 0 
[Conjured Mana Cake, 11, 1] => Conjured Mana Cake, 10, 0 
[Conjured Mana Cake, 11, -1] => Conjured Mana Cake, 10, 0 
[Conjured Mana Cake, 11, 49] => Conjured Mana Cake, 10, 47 
[Conjured Mana Cake, 11, 50] => Conjured Mana Cake, 10, 48 
[Conjured Mana Cake, 11, 55] => Conjured Mana Cake, 10, 53 
[Conjured Mana Cake, -1, 0] => Conjured Mana Cake, -2, 0 
[Conjured Mana Cake, -1, 1] => Conjured Mana Cake, -2, 0 
[Conjured Mana Cake, -1, -1] => Conjured Mana Cake, -2, 0 
[Conjured Mana Cake, -1, 49] => Conjured Mana Cake, -2, 45 
[Conjured Mana Cake, -1, 50] => Conjured Mana Cake, -2, 46 
[Conjured Mana Cake, -1, 55] => Conjured Mana Cake, -2, 51 
